### ISSUE TITLE
[Bug]-[30000]:Query Response screen table misaligned when screen height increased

### ISSUE THREAD

**Issue by btsgh** [appsmithorg/appsmith#13406]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I increase the height of the query response screen for say a postgresql query, the last column border of the table in the response page is not aligned correctly.

### Steps To Reproduce

1. Create a datasource for any PostGreSql DB. Get credentials from: https://www.notion.so/Servers-and-Credentials-665f00d9505745d59db4f3b4eda27c79#cd6a1610a7e84073b0491cd2c020de4e
2. 
<img width="644" alt="Screenshot 2022-04-28 at 1 01 26 PM" src="https://user-images.githubusercontent.com/101863839/165702406-39afcf65-c528-402b-9048-17b6ac34b64a.png">

3. Create a simple query: SELECT  gender, latitude, longitude FROM users ORDER BY id LIMIT 10;
4. Run it. When the mouse pointer turns to a divider above the Response / Errors tabs, drag the bottom screen to close the query, right below the query name. 
5. The last column table border is not aligned correctly.
<img width="559" alt="query_response_tablealignment" src="https://user-images.githubusercontent.com/101863839/165702853-b147f27d-ce67-45e2-a074-38d43e586073.png">

### Version

Release

**Comment by rohan-arthur** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 6000 |
| Effort (months)  | 0.1 |

**Comment by sneha122** [comment]
This issue is reproducible when the table does not have a scroll bar, when table has scroll bar, the header and table body is correctly aligned. After investigating this issue we found that, we have used **FixedSizeList** component of **react-window** library to display data in table, when there is more data in table and scroll bar appears, scrollbar width takes approximately 6px from the total table width, thus leading to misalignment of table's header and body, so in order to fix this issue pawan had added scrollbar width offset in FixedSizeList, this fixed the issue of header and body misalignment when scrollbar appears, but this caused an issue when scroll isn't there. Ideally this issue should have been handled by the FixedSizeList component itself which we are taking from react-window Library. We have two solutions to fix this problem from our end: 

- We can specify this scrollbar width offset based on the condition that if scrollbar is there, then we need to add this offset in the FixedSizeList's width, if scrollbar is not there then we do not add this offset width of scrollbar
- We can specify the styling for FixedSizeList and add `overflow: "overlay"` so that the scrollbar does not take extra width but rather appears on the table directly, but this solution would not work in firefox, safari browsers as this property is compatible with these browsers: ![Screenshot 2022-08-22 at 6.03.18 PM.png](https://images.zenhubusercontent.com/62ecb091669a591a4c1842c1/e6b273d1-415d-4559-b25f-dc8041ed24e6)

First solution looks more accurate.
